src/GraphQL/Resolvers/FuncFieldResolver.cs

```diff
@@ -104,12 +105,6 @@ public FuncFieldResolver(Func<IResolveFieldContext<TSourceType>, TReturnType?> r
                     var adapter = System.Threading.Interlocked.Exchange(ref _sharedAdapter, null);
                     adapter = adapter == null ? new ResolveFieldContextAdapter<TSourceType>(context) : adapter.Set(context);
                     var ret = resolver(adapter);
```

Comment:
This code really should never execute.  For instance, if the resolver returned `Task` (a void result), which makes no sense.  It is possible that someone derived from `Task<TResult>` and returned that here.  So for this code flow, since we can't anticipate the use case, re-using the adapter isn't a good idea.  We just don't know what value was really returned.  I really doubt this code will ever execute, however.

Reply:
You know better.

Reply:
I totally misread and botched this code.  Good thing we have tests for such sheer stupidity.